bin/common.c

```diff
@@ -57,6 +57,74 @@ static uint8_t unsafe_verify_host_fn(const char *host_name, size_t host_name_len
     return 1;
 }
 
+int write_array_to_file(const char *path, uint8_t *data, size_t length) {
+    GUARD_EXIT_NULL(path);
+    GUARD_EXIT_NULL(data);
+
+    FILE *file = fopen(path, "wb");
+    if (!file) {
+       return S2N_FAILURE;
+    }
+
+    if(fwrite(data, sizeof(char), length, file) != length) {
+        fclose(file);
+        return S2N_FAILURE;
+    }
+    fclose(file);
+
+    return S2N_SUCCESS;
+}
+
+int get_file_size(const char* path, size_t *length)
+{
+    GUARD_EXIT_NULL(path);
+    GUARD_EXIT_NULL(length);
+
+    FILE *file = fopen(path, "rb");
+    if (!file) {
+       return S2N_FAILURE;
+    }
+
+    if(fseek(file, 0, SEEK_END) < 0) {
```

Comment:
```suggestion
    if(fseek(file, 0, SEEK_END) != 0) {
```

Reply:
Good catch. I wish the file-reading C functions were more uniform with their return types.